library/src/main/java/com/novoda/downloadmanager/DownloadErrorFactory.java

```diff
@@ -41,8 +41,8 @@ static DownloadError createCannotWriteToFileError(DownloadFileStatus downloadFil
         return new DownloadError(DownloadError.Type.FILE_CANNOT_BE_WRITTEN, cannotWriteToFileMessage);
     }
 
-    static DownloadError createNetworkError(String networkErrorCause) {
-        String networkErrorMessage = "Network error, cannot download file. Cause: " + networkErrorCause;
+    static DownloadError createNetworkError(FileDownloader.FileDownloadError networkErrorCause) {
+        String networkErrorMessage = "Network error, cannot download file. Cause: " + networkErrorCause.message();
         return new DownloadError(DownloadError.Type.NETWORK_ERROR_CANNOT_DOWNLOAD_FILE, networkErrorMessage);
```

Comment:
This method reads as so:
- we are passed an error
- we get the msg out the error
- we make a new error of type NETWORK_ERROR_CANNOT_DOWNLOAD_FILE using the mesage

With the new change, we are passed the actual error, is there any point in redirecting that to create another error? i.e. could this whole method now be inlined? It could be potentially confusing to have two public api facing error types, that from here seem to have the same function? Perhaps thats the next step

Reply:
It's the next step 😄 Just trying to make small changes that I can do within a timeframe, otherwise I end up losing the context of what I am doing.

Reply:
next step, ok 👍 

not sure its fair to say a small PR helps keep you context, small commits will help and a small tick list for you to tick off does. But the PR is for other people to understand and review the changes, perhaps a different scope for the PR i.e. after a bit more work, would allow others to see more context of the change - or alternatively leaving a TODO in the code here so that you know what the next refactoring step is and reviewers can see this is going to change. Right now if all worked stopped after this merge, its a bad place.

not asking for changes here, just for thought 👍

Reply:
I see your point, I'll try and breakdown the steps a bit more and try and document them somewhere 😄 Normally I spend a great deal of time experimenting before picking an action but it might be better to stop and think, rather than just diving in.